server/monitor/src/main/java/org/apache/accumulo/monitor/rest/JettyMonitorApplication.java

```diff
@@ -0,0 +1,89 @@
+ */
+package org.apache.accumulo.monitor.rest;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.server.AccumuloServerContext;
+import org.apache.accumulo.server.client.HdfsZooInstance;
+import org.apache.accumulo.server.conf.ServerConfigurationFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.glassfish.jersey.filter.LoggingFilter;
+import org.glassfish.jersey.jackson.JacksonFeature;
+import org.glassfish.jersey.jetty.JettyHttpContainerFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.server.ServerProperties;
+import org.glassfish.jersey.server.mvc.MvcFeature;
+import org.glassfish.jersey.server.mvc.freemarker.FreemarkerMvcFeature;
+import org.glassfish.jersey.servlet.ServletProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JettyMonitorApplication extends MonitorApplication {
+  private static final Logger log = LoggerFactory.getLogger(JettyMonitorApplication.class);
+
+  public JettyMonitorApplication() {}
+
+  public void run() {
+    Instance instance = HdfsZooInstance.getInstance();
+    ServerConfigurationFactory config = new ServerConfigurationFactory(instance);
+    AccumuloServerContext serverContext = new AccumuloServerContext(config);
+
+    // Set the objects on the old monitor and start daemons to regularly poll the data
+    startDataDaemons(config, instance, serverContext);
+
+    final ResourceConfig rc = new ResourceConfig().register(FreemarkerMvcFeature.class).property(MvcFeature.TEMPLATE_BASE_PATH, "/templates")
+        .packages("org.apache.accumulo.monitor.rest.api", "org.apache.accumulo.monitor.rest.resources", "org.apache.accumulo.monitor.rest.view")
+        .property(ServerProperties.TRACING, "ALL").property(ServletProperties.FILTER_FORWARD_ON_404, "true")
+        .property(ServletProperties.FILTER_STATIC_CONTENT_REGEX, "/web/.*")
+        .register(new LoggingFilter(java.util.logging.Logger.getLogger("JettyMonitorApplication"), true)).register(JacksonFeature.class)
+        .registerClasses(AccumuloExceptionMapper.class);
+
+    final URI serverUri = getServerUri();
+
+    Server server = JettyHttpContainerFactory.createServer(serverUri, rc);
+
+    int port = serverUri.getPort();
+    if (0 == port) {
+      port = ((ServerConnector) server.getConnectors()[0]).getLocalPort();
+    }
+
+    String hostname = serverUri.getHost();
+
+    log.info("Server bound to " + hostname + ":" + port);
+
+    try {
+      advertiseHttpAddress(serverContext.getConnector().getInstance(), hostname, port);
+    } catch (AccumuloException | AccumuloSecurityException e) {
+      throw new RuntimeException("Failed to connect to Accumulo", e);
+    }
+
+    try {
+      System.in.read();
```

Comment:
I didn't do much work on this package as a whole since I think most of it will change when we substitute the current monitor with the REST one, I left this pretty much untouched since I wanted to make sure I had the REST calls first.